qiskit-ibm-transpiler-mcp-server/src/qiskit_ibm_transpiler_mcp_server/qta.py

```diff
@@ -62,12 +92,32 @@ async def _run_synthesis_pass(
         ai_synthesis_pass = PassManager(
             [synthesis_pass_class(backend=backend_service, **pass_kwargs)]
         )
-        loaded_quantum_circuit = load_qasm_circuit(circuit_qasm)
+        loaded_quantum_circuit = load_circuit(circuit, circuit_format=circuit_format)
         if loaded_quantum_circuit["status"] == "success":
-            ai_optimized_circuit = ai_synthesis_pass.run(loaded_quantum_circuit["circuit"])
+            original_circuit = loaded_quantum_circuit["circuit"]
+            original_metrics = _get_circuit_metrics(original_circuit)
+
+            ai_optimized_circuit = ai_synthesis_pass.run(original_circuit)
+            optimized_metrics = _get_circuit_metrics(ai_optimized_circuit)
+
+            # Return QPY format (source of truth for precision and chaining)
+            qpy_str = dump_circuit(ai_optimized_circuit, circuit_format="qpy")
+
+            # Calculate improvements (positive = reduction/improvement)
+            depth_reduction = original_metrics["depth"] - optimized_metrics["depth"]
+            two_qubit_reduction = (
+                original_metrics["two_qubit_gates"] - optimized_metrics["two_qubit_gates"]
+            )
+
             return {
                 "status": "success",
-                "optimized_circuit_qasm": dumps(ai_optimized_circuit),
+                "circuit_qpy": qpy_str,
+                "original_circuit": original_metrics,
+                "optimized_circuit": optimized_metrics,
+                "improvements": {
+                    "depth_reduction": depth_reduction,
+                    "two_qubit_gate_reduction": two_qubit_reduction,
```

Comment:
just curious since i am not too familiar, but could there be a case where either of these values are negative? and if so, would that have any impact on what is being returned here?